tmc-langs-r/src/main/java/fi/helsinki/cs/tmc/langs/r/RPlugin.java

```diff
@@ -0,0 +1,158 @@
+package fi.helsinki.cs.tmc.langs.r;
+
+
+import fi.helsinki.cs.tmc.langs.AbstractLanguagePlugin;
+import fi.helsinki.cs.tmc.langs.abstraction.ValidationResult;
+import fi.helsinki.cs.tmc.langs.domain.ExerciseBuilder;
+import fi.helsinki.cs.tmc.langs.domain.ExerciseDesc;
+import fi.helsinki.cs.tmc.langs.domain.RunResult;
+import fi.helsinki.cs.tmc.langs.domain.TestDesc;
+import fi.helsinki.cs.tmc.langs.io.StudentFilePolicy;
+import fi.helsinki.cs.tmc.langs.io.sandbox.StudentFileAwareSubmissionProcessor;
+import fi.helsinki.cs.tmc.langs.io.zip.StudentFileAwareUnzipper;
+import fi.helsinki.cs.tmc.langs.io.zip.StudentFileAwareZipper;
+
+import fi.helsinki.cs.tmc.langs.utils.ProcessRunner;
+
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.SystemUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Locale;
+
+public final class RPlugin extends AbstractLanguagePlugin {
+
+    /*
+    R folder contains the actual R files used in the
+    project/package. It is automatically included when creating a
+    R package but now when making a regular project in RStudio.
+     */
+    private static final Path R_FOLDER_PATH = Paths.get("R");
+
+    /*
+    test/testthat folder contains the unit testing
+    files which use the testThat library for the R project.
+    */
+    private static final Path TEST_FOLDER_PATH = Paths.get("tests");
+    private static final Path TESTTHAT_FOLDER_PATH = Paths.get("testthat");
+    private static final Path TESTTHAT_FILE_PATH = Paths.get("testthat.R");
+
+    private static final String CANNOT_RUN_TESTS_MESSAGE = "Failed to run tests.";
+    private static final String CANNOT_PARSE_TEST_RESULTS_MESSAGE = "Failed to read test results.";
+    private static final String CANNOT_SCAN_EXERCISE_MESSAGE = "Failed to scan exercise.";
+    private static final String CANNOT_PARSE_EXERCISE_DESCRIPTION_MESSAGE =
+            "Failed to parse exercise description.";
+
+    private static Logger log = LoggerFactory.getLogger(RPlugin.class);
+
+    public RPlugin() {
+        super(
+                new ExerciseBuilder(),
+                new StudentFileAwareSubmissionProcessor(),
+                new StudentFileAwareZipper(),
+                new StudentFileAwareUnzipper());
+    }
+
+    /*
+    NOTE: Files.exists does not seem to be able to verify the R and
+    testthat folder's existence if they are empty.
```

Comment:
You could try to reverse the logic via use of `!Files.notExists`, since that deals with unclear cases differently. But that same change in the behaviour in unclear cases might also end up just causing more issues... Not sure how to approach this.

Reply:
A further question, does this issue manifest only on some certain OS or all?

Have you checked if [this](https://stackoverflow.com/a/3199361) is related?

Reply:
An even further suggestion: check if `Files.isDirectory` solves the issue.

Reply:
Sounds fair, no need to spend time on inconsequential.

I also think we store all assignments as git repositories, so empty folders in assignment templates are not a thing anyways.